ambry-clustermap/src/main/java/com.github.ambry.clustermap/HelixClusterAggregator.java

```diff
@@ -0,0 +1,146 @@
+/**
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package com.github.ambry.clustermap;
+
+import com.github.ambry.server.StatsSnapshot;
+import com.github.ambry.server.StatsWrapper;
+import com.github.ambry.utils.Pair;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import org.codehaus.jackson.annotate.JsonAutoDetect;
+import org.codehaus.jackson.map.ObjectMapper;
+
+
+/**
+ * Responsible for performing cluster wide stats aggregation.
+ */
+public class HelixClusterAggregator {
+  private final ObjectMapper mapper = new ObjectMapper();
+  private final long relevantTimePeriodInMs;
+
+  HelixClusterAggregator(long relevantTimePeriodInMinutes) {
+    relevantTimePeriodInMs = TimeUnit.MINUTES.toMillis(relevantTimePeriodInMinutes);
+    mapper.setVisibilityChecker(mapper.getVisibilityChecker().withFieldVisibility(JsonAutoDetect.Visibility.ANY));
+  }
+
+  /**
+   * Take a {@link List} of JSON string representation of {@link StatsWrapper} objects and perform cluster wide
+   * aggregation with them.
+   * @param statsWrappersJSON a {@link List} of JSON string representation of {@link StatsWrapper} objects from the
+   *                          node level
+   * @return a {@link Pair} of Strings whose first element is the raw (sum) aggregated stats and whose second element is
+   * the aggregated stats based on timestamp and value from each partition.
+   * @throws IOException
+   */
+  Pair<String, String> doWork(List<String> statsWrappersJSON) throws IOException {
+    StatsSnapshot partitionSnapshot = new StatsSnapshot(0L, new HashMap<String, StatsSnapshot>());
+    Map<String, Long> partitionTimestampMap = new HashMap<>();
+    StatsSnapshot rawPartitionSnapshot = new StatsSnapshot(0L, new HashMap<String, StatsSnapshot>());
+    for (String statsWrapperJSON : statsWrappersJSON) {
+      if (statsWrapperJSON != null) {
+        StatsWrapper snapshotWrapper = mapper.readValue(statsWrapperJSON, StatsWrapper.class);
+        StatsWrapper snapshotWrapperCopy = mapper.readValue(statsWrapperJSON, StatsWrapper.class);
+        combineRaw(rawPartitionSnapshot, snapshotWrapper);
+        combine(partitionSnapshot, snapshotWrapperCopy, partitionTimestampMap);
+      }
+    }
+    StatsSnapshot reducedRawSnapshot = reduce(rawPartitionSnapshot);
+    StatsSnapshot reducedSnapshot = reduce(partitionSnapshot);
```

Comment:
what is the format of this reduced snapshot? Does it have partition info or is it removed?